### PR TITLE
Add PARALLEL support

### PR DESCRIPTION
**Pull request by Algunenano**
Add PARALLEL and VOLATILE tags for Postgresql functions

### CODE REVIEW

**Comment by Algunenano** [comment]
After some final fixes I think this should be done with the PR.

**Comment by Algunenano** [comment]
Solves https://github.com/CartoDB/crankshaft/issues/181

File: src/pg/sql/cdb_utils.sql [hunk 1]
```diff
@@ -299,7 +299,7 @@ BEGIN
 
     RETURN (best_result)[2:array_upper(best_result, 1)];
 END;
-$$ language plpgsql IMMUTABLE;
```
Comment (rafatower) [inline src/pg/sql/cdb_utils.sql]: Are you sure this is `VOLATILE`? the code seems to at least try to provide reproducibility (`setseed` above). Any side effects maybe?

Comment (Algunenano) [reply src/pg/sql/cdb_utils.sql]: Technically both setseed and random are volatile

```
analysis_api_test_db=# \df+ setseed
                                                                                   List of functions
   Schema   |  Name   | Result data type | Argument data types |  Type  | Volatility |  Parallel  |  Owner   | Security | Access privileges | Language | Source code |   Description   
------------+---------+------------------+---------------------+--------+------------+------------+----------+----------+-------------------+----------+-------------+-----------------
 pg_catalog | setseed | void             | double precision    | normal | volatile   | restricted | postgres | invoker  |                   | internal | setseed     | set random seed
(1 row)

analysis_api_test_db=# \df+ random 
                                                                                 List of functions
   Schema   |  Name  | Result data type | Argument data types |  Type  | Volatility |  Parallel  |  Owner   | Security | Access privileges | Language | Source code | Description  
------------+--------+------------------+---------------------+--------+------------+------------+----------+----------+-------------------+----------+-------------+--------------
 pg_catalog | random | double precision |                     | normal | volatile   | restricted | postgres | invoker  |                   | internal | drandom     | random value
```
The issue here is that modifying the seed could be considered as part of the output as that is not cacheable. For performance reasons we could consider setting it as STABLE as done in the quota/redis calls.

**Comment by Algunenano** [comment]
> I'd be nice to copy the volatility and "parallelizability" guidelines in the CONTRIBUTING.md as well here.
Will do

File: src/pg/sql/cdb_utils.sql (hunk 1 above)
Comment (rafatower) [reply src/pg/sql/cdb_utils.sql]: hm, you're right, a call to `setseed` provokes a side effect (and global state change, namely the random number generator) that prevents it from being parallel safe.

**Comment by Algunenano** [comment]
Rebased with fix for bsd-systems compatibility (as https://github.com/CartoDB/cartodb-postgresql/pull/315)

**Comment by Algunenano** [comment]
Rebased on top of GWR introduction and added categories to the new functions.

It'd nice if we could merge this and deploy it with GWR so we don't have to do 2 different deploys. cc/ @andy-esch @jgoizueta 
Related to this, a warning has appeared when running python tests: https://github.com/CartoDB/crankshaft/issues/188

**Comment by Algunenano** [comment]
After checking the deploy method with @jgoizueta I''ve seen that this projects commits release files (cartodb-postgresql doesn't) so instead of stripping the PARALLEL labels when generating the release file we need to do it in the deploy phase. Should be ok with this last commit.

Also we've decided to do 2 different deploys.

**Comment by rafatower** [comment]
:+1:

**Pull request merged**